end_to_end_tests/golden-record/my_test_api_client/models/model_with_union_property.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict, Union
+
+import attr
+
+from ..models.an_enum import AnEnum
+from ..models.an_int_enum import AnIntEnum
+from ..types import UNSET, Unset
+
+
+@attr.s(auto_attribs=True)
+class ModelWithUnionProperty:
+    """  """
+
+    a_property: Union[Unset, AnEnum, AnIntEnum] = UNSET
+
+    def to_dict(self) -> Dict[str, Any]:
+        a_property: Union[Unset, AnEnum, AnIntEnum]
+        if isinstance(self.a_property, Unset):
+            a_property = UNSET
+        elif isinstance(self.a_property, AnEnum):
+            a_property = UNSET
+            if not isinstance(self.a_property, Unset):
+                a_property = self.a_property
```

Comment:
This is weird, I'll try to find the place in the template that causes it. Obviously no need to double check `isinstance` on the same property, we already know it's `AnEnum`.

Reply:
I don't see where your changes would have caused this, so it was probably an existing issue with Unions/UNSET (I know there is at least one issue related to that). The inner_property is probably being set as not required which makes the inner constructor redo the check. I can handle that in a future version.